common/src/character_controller.rs

```diff
@@ -149,10 +179,49 @@ impl CharacterControllerPass<'_> {
     }
 }
 
+/// Modifies the `subject` by a linear combination of the `normals` to ensure that it is approximately
+/// orthogonal to all the normals. The normals are assumed to be linearly independent, and, assuming the final
+/// result is nonzero, a small correction is applied to ensure that the subject is moving away from the surfaces
+/// the normals represent even when floating point approximation is involved.
```

Comment:
What if the character is moving along a concave curve? I think that could produce an arbitrarily large number of linearly dependent vectors.

Reply:
Each step of the concave curve pushes the player away from the previous normal, so only the most recent normal gets applied. This function then only gets called with one normal, satisfying its constraint.

Reply:
Ah, missed the `retain`.